pkg/prop/prop.go

```diff
@@ -145,11 +145,10 @@ func (p *MediaConstraints) FitnessDistance(o Media) (float64, bool) {
 	cmps.add(p.Width, o.Width)
 	cmps.add(p.Height, o.Height)
 	cmps.add(p.FrameFormat, o.FrameFormat)
-	// The next line is comment out for now to not include framerate in the fitness function.
-	// As camera.Properties does not have access to the list of available framerate at the moment,
-	// no driver can be matched with a framerate constraint.
-	// Note this also affect screen caputre as screen.Properties does not fill in the Framerate field.
-	// cmps.add(p.FrameRate, o.FrameRate)
+	// skip framerate if not available in media properties
+	if o.FrameRate > 0.0 {
```

Comment:
What are the consequences of relying the logic in the second bullet? 

I'm guessing for Mac OS X, since the framerate isn't currently supported in this lib everything will be penalized by the same amount and we'll still prioritize drivers the same? 

For Linux it seems like requesting a framerate and for some reason not being able to retrieve one would penalize the props just as much as having a wrong framerate which seems fine.

To do the first bullet you'd have to do something like...

```
type FrameRateConstraint interface {
	Compare(frame.Format) (float64, bool)
	Value() (frame.Format, bool)
}

type Float float32

func (f Float) Compare(a float32) (float64, bool) {
	if a < 0.0 {
		return 0.0, true
	}
	return FloatExact(float32(f)).Compare(a)
}
...

```
Which isn't too bad but might be overkill if the `FloatConstraint` logic works as is.

Reply:
1. FrameRateConstraint

This seems like a decent option. The downside is increasing the surface area of prop code. The implementation would be the exact same as `FloatConstraint` with unset checks, which seems redundant. Perhaps there is a nice golang way to wrap FloatConstraint?

It seems much simpler to add a one-liner check in FitnessDistance.

2. Allow unset FPS props

`selectBestDriver` would still function properly. The only potential issue would be for users who use `MediaConstraints.FitnessDistance` directly as there would be a breaking change to fitness scores. 

I am assuming this is a rare case, as most users are going through the `GetUserMedia`/`selectVideo` endpoints. Also scores would only be affected if FrameRate is specified in the `MediaStreamConstraints.`

~Note: Only relavant when FrameRate is